clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -4694,9 +4695,9 @@ def note_protocol_decl : Note<
   "protocol is declared here">;
 def note_protocol_decl_undefined : Note<
   "protocol %0 has no definition">;
-def err_attribute_preferred_name_arg_invalid : Error<
-  "argument %0 to 'preferred_name' attribute is not a typedef for "
-  "a specialization of %1">;
+def err_attribute_preferred_name_arg_invalid
```

Comment:
Same here, how about `err_attribute_not_typedef_for_specialization`? (I'm not tied to any of these names, so if you have a better suggestion, go with it.)

Reply:
@AaronBallman I've updated both messages